imports/plugins/core/orders/client/containers/invoiceContainer.js

```diff
@@ -39,44 +54,562 @@ class InvoiceContainer extends Component {
+            if (error) {
+              Alerts.alert(error.reason);
+            }
+            if (result) {
+              Alerts.toast(i18next.t("mail.alerts.emailSent"), "success");
+            }
+            this.setState({
+              isRefunding: false
+            });
+          });
+        }
+      });
+    }
+  }
+
+  render() {
     return (
       <TranslationProvider>
         <Invoice
-          canMakeAdjustments={canMakeAdjustments}
-          paymentCaptured={paymentCaptured}
-          isOpen={this.state.isOpen}
-          discounts={discounts}
+          {...this.props}
+
+          getSelectedItemsInfo={this.getSelectedItemsInfo}
           handleClick={this.handleClick}
-          invoice={invoice}
-          orderId={orderId}
-          refunds={refunds}
-          dateFormat={this.dateFormat}
-          isFetching={isFetching}
-          collection={collection}
-          hasRefundingEnabled={hasRefundingEnabled}
+          handleSelectAllItems={this.handleSelectAllItems}
+          onClose={this.handleClose}
+          togglePopOver={this.togglePopOver}
+          handleInputChange={this.handleInputChange}
+          handleItemSelect={this.handleItemSelect}
+          displayMedia={this.handleDisplayMedia}
+          toggleUpdating={this.toggleUpdating}
+          handleReturnItems={this.handleReturnItems}
+          getRefundedItemsInfo={this.getRefundedItemsInfo}
+          handleApprove={this.handleApprove}
+          isAdjusted={this.isAdjusted}
+          handleCapturePayment={this.handleCapturePayment}
+          handleRefund={this.handleRefund}
+          hasRefundingEnabled={this.hasRefundingEnabled()}
+
+          isOpen={this.state.isOpen}
+          refunds={this.state.refunds}
+          isCapturing={this.state.isCapturing}
+          selectAllItems={this.state.selectAllItems}
+          selectedItems={this.state.selectedItems}
+          currency={this.state.currency}
+          isRefunding={this.state.isRefunding}
+          popOverIsOpen={this.state.popOverIsOpen}
+          editedItems={this.state.editedItems}
+          isUpdating={this.state.isUpdating}
         />
       </TranslationProvider>
     );
   }
 }
 
+// helper to return the order payment object
+// the first credit paymentMethod on the order
+// returns entire payment method
+function orderCreditMethod(order) {
+  return order.billing.filter(value => value.paymentMethod.method ===  "credit")[0];
+}
+
 const composer = (props, onData) => {
+  const order = props.order;
+  const refunds = props.refunds;
+
+  const paymentMethod = orderCreditMethod(order).paymentMethod;
+  const orderStatus = orderCreditMethod(order).paymentMethod.status;
+  const orderMode = orderCreditMethod(order).paymentMethod.mode;
+  const orderDiscounts = orderCreditMethod(order).invoice.discounts;
+
+  const paymentApproved = order.billing[0].paymentMethod.status === "approved";
+  const paymentCaptured = orderStatus === "completed" ||
```

Comment:
I wonder if checking the value against an array of values wouldn't be better than all these chained conditionals. e.g.
`const paymentPendingApproval = _.includes(["created", "adjustments", "error"], orderStatus)` or something to that effect.